client/my-sites/checkout/checkout/index.jsx

```diff
@@ -541,14 +547,25 @@ export class Checkout extends React.Component {
 
 		this.setDestinationIfEcommPlan( pendingOrReceiptId );
 
-		// if it is one of the Jetpack products, use product info as a parameter.
-		// We want to be product-specific, as the same path will likely be used for
-		// WP.com sites purchasing Search.
-		if (
-			startsWith( product, 'jetpack_backup' ) ||
-			startsWith( product, 'jetpack_search' ) ||
-			startsWith( product, 'jetpack_scan' )
-		) {
+		// If it is a Jetpack Cloud products, use the redirection lib.
+		// For other Jetpack products, use the fallback logic to send to Calypso.
+		if ( isJetpackCloudProductSlug( product ) && isJetpackNotAtomic ) {
+			let source = '';
+			if ( isJetpackBackupSlug( product ) ) {
+				source = 'calypso-backups';
+			} else if ( isJetpackScanSlug( product ) ) {
+				source = 'calypso-scanner';
+			}
+			if ( source ) {
+				return addQueryArgs(
+					{
+						source,
+						site: selectedSiteSlug,
```

Comment:
Will the Jetpack redirects work just the same with a site slug versus a blog id?

Reply:
Yep! They did in my testing, at least.